assets/src/components/map.tsx

```diff
@@ -227,88 +229,66 @@ const EventAdder = ({
 }
 
 // #region Auto Center Functionality
-export const autoCenter = (
-  map: LeafletMap,
-  latLngs: LatLngExpression[],
-  pickerContainerIsVisible: boolean
-): void => {
-  if (latLngs.length === 0) {
-    map.setView(defaultCenter, 13)
-  } else if (latLngs.length === 1) {
-    map.setView(latLngs[0], 16)
-  } else if (latLngs.length > 1) {
-    map.fitBounds(Leaflet.latLngBounds(latLngs), {
-      paddingBottomRight: [20, 50],
-      paddingTopLeft: [pickerContainerIsVisible ? 220 : 20, 20],
-    })
-  }
+export type UpdateMapFromPointsFn = (map: LeafletMap, points: LatLng[]) => void
+
+export interface FollowerProps {
+  positions: LatLng[]
+  onUpdate: UpdateMapFromPointsFn
 }
 
-const AutoCenter = ({
-  latLngs,
-  shouldAutoCenter,
-  isAutoCentering,
-}: {
-  shouldAutoCenter: boolean
-  isAutoCentering: MutableRefObject<boolean>
-  latLngs: LatLng[]
-}) => {
+export const Follower = ({
+  positions,
+  isUpdating,
+  onUpdate,
+  shouldFollow = true,
+}: FollowerProps & InteractiveFollowState) => {
   const map = useMap()
-
-  const [{ pickerContainerIsVisible }] = useContext(StateDispatchContext)
-  const [currentLatLngs, setCurrentLatLngs] = useState<LatLng[]>(latLngs)
+  const [currentLatLngs, setCurrentLatLngs] = useState<LatLng[]>(positions)
 
   if (
-    !equalByElements(latLngs, currentLatLngs, (latLng1, latLng2) =>
-      latLng1.equals(latLng2)
-    )
+    !equalByElements(positions, currentLatLngs, (lhs, rhs) => lhs.equals(rhs))
   ) {
-    setCurrentLatLngs(latLngs)
+    setCurrentLatLngs(positions)
   }
 
   useEffect(() => {
-    if (map !== null && shouldAutoCenter) {
-      isAutoCentering.current = true
-      autoCenter(map, currentLatLngs, pickerContainerIsVisible)
+    if (map !== null && shouldFollow) {
+      if (isUpdating !== undefined) {
+        isUpdating.current = true
+      }
+      onUpdate(map, currentLatLngs)
     }
-  }, [
-    map,
-    shouldAutoCenter,
-    isAutoCentering,
-    currentLatLngs,
-    pickerContainerIsVisible,
-  ])
+  }, [map, shouldFollow, isUpdating, currentLatLngs, onUpdate])
 
   return <></>
 }
+export interface InteractiveFollowState {
+  isUpdating: MutableRefObject<boolean>
```

Comment:
nitpick (non-blocking): The name `isUpdating` was a bit confusing to me at first with the combination of the name `InteractiveFollowState` - I didn't quite get what it is that is being updated, in combination with it being set to `isFollowing` on line 280 below.

Reply:
thank you!

Reply:
let me know what you think of 7be135c4013b264282c75fe8618aca76dff3e0d7 ?

Reply:
Thank you - `isAnimatingFollowUpdate` is so clear!!